test/common/upstream/load_balancer_fuzz.h

```diff
@@ -0,0 +1,53 @@
+#include "common/upstream/load_balancer_impl.h"
+
+#include "test/common/upstream/load_balancer_fuzz.pb.validate.h"
+#include "test/common/upstream/load_balancer_fuzz_test_utils.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class LoadBalancerFuzzBase : public LoadBalancerFuzzTestBase {
+public:
+  LoadBalancerFuzzBase();
+
+  // Untrusted upstreams don't have the ability to change the host set size, so keep it constant
+  // over the fuzz iteration.
+  void initializeFixedHostSets(uint32_t num_hosts_in_priority_set,
+                               uint32_t num_hosts_in_failover_set);
+
+  virtual void initialize(test::common::upstream::LoadBalancerTestCase input) PURE;
+  void initializeAndReplay(test::common::upstream::LoadBalancerTestCase input);
+  void updateHealthFlagsForAHostSet(bool failover_host_set, uint32_t num_healthy_hosts,
+                                    uint32_t num_degraded_hosts = 0,
+                                    uint32_t num_excluded_hosts = 0);
+  // These two actions have a lot of logic attached to them such as mocks, so you need to delegate
+  // these to specific logic per each specific load balancer. This makes sense, as the load
+  // balancing algorithms sometimes use other components which are tightly coupled into the
+  // algorithm logically.
+  virtual void prefetch() PURE;
+  virtual void chooseHost() PURE;
+  virtual ~LoadBalancerFuzzBase() = default;
+
+private:
+  void replay(test::common::upstream::LoadBalancerTestCase input);
+  // TODO: Move load balancer here?
+};
+
+class RandomLoadBalancerFuzzTest : public LoadBalancerFuzzBase {
+public:
+  RandomLoadBalancerFuzzTest();
+  void initialize(test::common::upstream::LoadBalancerTestCase input) override;
+  // Has mock logic for random
+  void prefetch() override;
+  void chooseHost() override;
+  ~RandomLoadBalancerFuzzTest() = default;
+
+  std::unique_ptr<RandomLoadBalancer> load_balancer_;
+};
+
+// TODO: Since each load balancer has a specific config, this will need to be addressed in their
```

Comment:
if LB might need a lot of specific protos, it might point to having specific TestCase inputs for each LB

message SubsetLoadBalancerTestCase {
    SubsetLbConfig subset_config;
    test.fuzz.load_balancer_base.CommonConfigurations common_configs;
    test.fuzz.load_balancer_base.LbActions actions;
}

Reply:
Hey, this is exactly what I was planning! Right now, I'm working on the load balancers that don't take a seperate config in, and only use common. This kind of maps to my high level workflow of first, taking care of codepaths tested in https://github.com/envoyproxy/envoy/blob/master/test/common/upstream/load_balancer_impl_test.cc, starting with random (your great suggestion), then moving onto round robin, then weighted least request, then the base class, then the more specific seperate maglev and ring hash. This is why my design is so unfleshed out currently, and the design only maps to the load balancers which don't take in a seperate config. However, the future design decisions are logically orthogonal and will be extensions to the current design?